cmd/precise-code-intel-api-server/internal/api/observability.go

```
@@ -20,14 +21,25 @@ type ObservedCodeIntelAPI struct {
 
 var _ CodeIntelAPI = &ObservedCodeIntelAPI{}
 
-// NewObservedCodeIntelAPI wraps the given CodeIntelAPI with error logging, Prometheus metrics, and tracing.
-func NewObserved(codeIntelAPI CodeIntelAPI, observationContext *observation.Context) CodeIntelAPI {
-	metrics := metrics.NewOperationMetrics(
+// createOperationMetrics creates the metric instance for all operations in an ObservedCodeIntelAPI.
+// No metrics are created if the given registerer is nil.
+func createOperationMetrics(r prometheus.Registerer) *metrics.OperationMetrics {
+	if r == nil {
+		return nil
```

Why do you want this nil check? Shouldn’t the caller not invoke the method? Otherwise this allows it to silently fail

---

I'll update it so tests always supply a no-op registerer.